sdks/java/build-tools/src/main/resources/docker/README.md

```diff
@@ -0,0 +1,180 @@
+
+    docker run \
+      -e URL="https://www.apache.org/dyn/closer.cgi?filename=beam/2.0.0/apache-beam-2.0.0-source-release.zip&action=download" \
+      -e SRC_FILE="apache-beam-2.0.0-source-release.zip" \
+      -e SRC_DIR="apache-beam-2.0.0" \
+      -it beam:openjdk8 /bin/bash
+
+If you want to run a container with a specific version (e,g, 2.1.0-RC1):
+
+    docker run \
+      -e URL="https://github.com/apache/beam/archive/v2.1.0-RC1.zip" \
+      -e SRC_FILE="v2.1.0-RC1.zip" \
+      -e SRC_DIR="beam-2.1.0-RC1" \
+      -it beam:openjdk8 /bin/bash
+
+*Notice that SRC_FILE is different from SRC_DIR because github replaces the 'v'
+character in the internal name of the zip*.
+
+To run a container with the source code during a vote:
+
+    docker run \
+      -e URL="https://dist.apache.org/repos/dist/dev/beam/2.1.0/apache-beam-2.1.0-source-release.zip" \
+      -e SRC_FILE="apache-beam-2.1.0-source-release.zip" \
+      -e SRC_DIR="apache-beam-2.1.0" \
+      -it beam:openjdk8 /bin/bash
+
+## Testing in an specific environment with your own source
+
+You can also overwrite the volume containing the source code of Beam from a
+directory in your host machine. This is useful to test your code in different
+versions of Java.
+
+If you have the code in the ~/workspace/beam directory and you want to quickly
+test it with Beam on Java 7 you can do:
+
+    docker run -v ~/workspace/beam:/home/user/beam -it beam:openjdk7 /bin/bash
+
+## Performance improvements
+
+The docker image does **not** contain the java dependencies and this could make
+the execution of the commands in the image take more time than expected. One way
+to speed this is by mounting a directory with the dependencies as a docker
+volume. The following command shows how to do this locally from the host machine
+mapping the local maven's `~/.m2` directory in the container.
+
+    docker run -v ~/.m2:/home/user/.m2 -it beam:openjdk8 /bin/bash
+
+You can also use an empty directory instead of `~/.m2` if you want to create a
+different environment from the local one to share it between runs.
+
+**Maven Improvements:**
+
+You can also speed the execution by adding environment variables to maven, for
+example the MAVEN_OPTS environment variable may speed up compilation:
+ 
+    docker run \
+      -e MAVEN_OPTS='-client -XX:+TieredCompilation -XX:TieredStopAtLevel=1 -Xverify:none' \
+      -it beam:openjdk8 /bin/bash
+
+Maven does not execute on multiple cores by default, to do this you have to
+enable the threading execution:
+    
+    mvn --threads 1C ...
+
+## Python
+
+The python version of the file container differs from the other versions because
+it only contains the python source code, if you want to build the python image
+based on the latest release you can run:
+
+    cd file/python2
+    docker build -t beam:python2 .
+
+You can then run the python tests or install it
+
+    python setup.py install --user
```

Comment:
This is fine. We normally recommend running a virtualenv which avoids the need for using `--user` flag but you are right otherwise. IMO, there is no security risk this will be a development container for users, we are not providing a production ready container here.

Unrelated comment, we might want to install gcp package (e.g. `pip install -e .[gcp]`)